### ISSUE TITLE
Problem validating JSON document against schema - nodejs agent

### ISSUE THREAD

**Issue by shadowlik** [elastic/apm-agent-nodejs#819]
Hi,

I don't know if this is a bug but I was encountering this error :

```
Problem validating JSON document against schema: I[#] S[#]doesn't validate with \"transaction#\"\n  I[#] S[#/allOf/0] allOf failed\n    I[#/context/request/url/port] S[#/allOf/0/properties/context/properties/request/properties/url/properties/port/type] expected string or null, but got number
```

I did some investigation and changed inside the function getUrlFromRequest in the elastic-apm-node/lib/parsers.js from:

``` javascript
// port
  if (url.port) result.port = Number(url.port)
  else if (req.headers['x-forwarded-port']) result.port = Number(getFirstHeader(req, 'x-forwarded-port'))
  else if (host.port) result.port = Number(host.port)
```
to:

``` javascript
  // port
  if (url.port) result.port = String(url.port)
  else if (req.headers['x-forwarded-port']) result.port = String(getFirstHeader(req, 'x-forwarded-port'))
  else if (host.port) result.port = String(host.port)
```
And now my APM agent is working fine...

Before I dove in the elastic-apm files, I tried to change my port to string and it didn't work.

I'm using:

* Elasticsearch 6.5.4
* Elasticsearch APM 6.5.4
* APM Agent 2.2.1

In my project I'm using typescript and only express...

**Comment by watson** [comment]
@shadowlik Thanks for reporting this. I'm sorry you got affected by this. We thought we had fixed this bug in v2.2.1, but it turns out it was still possible to trigger it. We have just released v2.3.0 with a proper fix.

**Comment by shadowlik** [comment]
Thanks @watson, for the quick response and fix... Just updated my applications with the new version and now everything working fine!
